### PR TITLE
Bug/issue 324 update python yaml

### PR DESCRIPTION
**Pull request by afsmythe**
There remain some issues with how the non CSV elements (`InternationalizedText`, for example) remain included in the CSV docs in this PR. AFAIK this is because the scripts expect a `csv-header-name` for each YAML item and Tag. So, I've added one for `InternationalizedText` to allow the make_rest command to complete. Ideally `csv-header-name` would be optional and only entered in the YAML for  CSV files and column headers, and those YAML _name tags without a `csv-header-name` would not be present within the CSV rst files. Didn't want to change the logic too much without gathering some suggestions on how best to handle this. 

Additionally, the XML `Department` _sub_type (of `ElectionAdministration`) is not compatible with the CSV `department`, which is [its own file](https://vip-specification.readthedocs.io/en/latest/csv/element_files/department.html) in the CSVs. `Department` is not given its own page in the CSV documentation in this PR because it is a _sub_type of `ElectionAdministration`, and it should be given its own page. 

I have an idea on how to correct this issue with Department - separating out the YAML item `_sub_types` in to `_sub_types_csv` and `_sub_types_xml`, but wanted to ask for more input first.

### CODE REVIEW

File: docs/yaml/elements/candidate_selection.yaml [hunk 1]
```diff
@@ -12,17 +24,25 @@ post: |-
...
   error_then: =must-ignore
+  repeats: true
   type: xs:IDREFS
 - _name: EndorsementPartyIds
+  csv-header-name: endorsement_party_ids
+  csv-type: xs:IDREFS
   description: References a set of :ref:`$$$-party` elements, which signifies one
     or more endorsing parties for the candidate(s).
   error_then: =must-ignore
+  repeats: true
```
Comment (jdmgoogle) [inline docs/yaml/elements/candidate_selection.yaml]: IDREFS means a single whitespace-separated list. So "repeats" would mean multiple whitespace-separated lists.

File: docs/yaml/elements/candidate_selection.yaml [hunk 2]
```diff
@@ -12,17 +24,25 @@ post: |-
      </CandidateSelection>
 tags:
 - _name: CandidateIds
+  csv-header-name: candidate_ids
+  csv-type: xs:IDREFS
   description: References a set of :ref:`$$$-candidate` elements. The number of candidates
     that can be references is unbounded in cases where the ballot selection is for
     a ticket (e.g. "President/Vice President", "Governor/Lt Governor").
   error_then: =must-ignore
+  repeats: true
```
Comment (jdmgoogle) [inline docs/yaml/elements/candidate_selection.yaml]: See other comment about IDREFS.

File: docs/yaml/elements/election.yaml [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+_sub_types:
+- InternationalizedText
+csv-header-name: election
+csv-post: |-
+  .. code-block:: csv-table
+     :linenos:
+
+
+      id,date,name,election_type,state_id,is_statewide,registration_info,absentee_ballot_info,results_uri,polling_hours,has_election_day_registration,registration_deadline,absentee_request_deadline,hours_open_id
+      e001,10-08-2016,Best Hot Dog,State,st51,true,www.registrationinfo.com,You can vote absentee,http://hotdogcontest.gov/results,Noon to 3p.m.,true,10/08/2016,,ho002
```
Comment (jdmgoogle) [inline docs/yaml/elements/election.yaml]: I believe you mean "Best Sandwich".

File: docs/yaml/elements/lat_lng.yaml [hunk 4]
```diff
@@ -21,16 +22,22 @@ post: |-
      </PollingLocation>
 tags:
 - _name: Latitude
+  csv-header-name: latitude
+  csv-type: xs:float
```
Comment (jdmgoogle) [inline docs/yaml/elements/lat_lng.yaml]: Once #375 goes in this will be xs:double.

File: docs/yaml/elements/lat_lng.yaml [hunk 5]
```diff
@@ -21,16 +22,22 @@ post: |-
...
   description: The latitude of the polling location.
   error_then: =must-ignore
   required: true
   type: xs:float
 - _name: Longitude
+  csv-header-name: longitude
+  csv-type: xs:float
```
Comment (jdmgoogle) [inline docs/yaml/elements/lat_lng.yaml]: Same comment re: #375.

File: docs/yaml/elements/locality.yaml [hunk 6]
```diff
@@ -21,21 +32,29 @@ post: |-
...
+  csv-header-name: name
+  csv-type: xs:string
   description: Specifies the name of a locality.
   error: If the field is not present or invalid, the implementation is required to
     ignore the Locality element containing it.
   required: true
   type: xs:string
 - _name: PollingLocationIds
+  csv-header-name: polling_location_ids
+  csv-type: xs:IDREF
```
Comment (jdmgoogle) [inline docs/yaml/elements/locality.yaml]: IDREF or IDREFS?

File: scripts/vippy/rest.py [hunk 7]
```diff
@@ -183,23 +212,37 @@ def update_rest_file(all_types, data_type, prefix):
...
         return
+    if data_type.is_extends:
+        _log.debug("skipping rest file for extended class: {0}".format(type_name))
+        return
 
-    rest_path = data_type.rest_path
-    _log.debug("updating: {0}".format(rest_path))
+    #rest_path = data_type.rest_path
+    rest_path_xml = data_type.rest_path_xml
+    #print(rest_path_xml)
```
Comment (jdmgoogle) [inline scripts/vippy/rest.py]: Debugging leftovers?

File: scripts/vippy/rest.py [hunk 8]
```diff
@@ -215,7 +258,8 @@ def update_rest_file_single_page(all_types):
         label = "{0}-{1}".format(prefix, title.lower())
         rest += make_rest_header(title, label=label, header_char="-")
         for data_type in data_types:
-            if data_type.is_sub_type:
+            if data_type.is_sub_type and data_type.is_extends:
+            #if data_type.is_sub_type:
```
Comment (jdmgoogle) [inline scripts/vippy/rest.py]: Debugging?

File: scripts/vippy/rest.py [hunk 9]
```diff
@@ -227,22 +271,26 @@ def update_rest_files(type_name=None):
...
-    prefix = "multi-xml"
-    all_types = common.get_all_types()
-    type_map = all_types.type_map
-
-    if type_name is None:
-        type_names = sorted(type_map.keys())
-    else:
-        type_names = [type_name]
+    for mode in ['csv','xml']:
+        #print("**** In Single Mode {}".format(mode))
```
Comment (jdmgoogle) [inline scripts/vippy/rest.py]: Debugging?

File: docs/csv.rst [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+	- office.txt
+
+Files to serve referenda and ballot measures:
+	- ballot_measure_contest.txt 
+	- ballot_measure_selection.txt
+
+Files to serve retention contests:
+	- retention_contest.txt
+
+CSV files must be comma-delimited, UTF-8 .txt files, named according to the specification. The id attribute
```
[the remark this answers is not in the document]
Comment (afsmythe) [reply docs/csv.rst]: Yea - presently we require files extended with ".txt". There isn't any reason for doing this and may be changed to also allow for ".csv" but for now the documentation is correct.

File: docs/yaml/elements/election.yaml (hunk 3 above)
Comment (afsmythe) [reply docs/yaml/elements/election.yaml]: Good catch :)

File: docs/yaml/elements/locality.yaml (hunk 6 above)
Comment (jdmgoogle) [reply docs/yaml/elements/locality.yaml]: Last issue, I believe.

Comment (afsmythe) [reply docs/yaml/elements/locality.yaml]: Woops. Thanks, missed this.

**Comment by afsmythe** [comment]
I believe all raised issues have been resolved. Free to merge?

**Comment by afsmythe** [comment]
Reminder that these commits will need to be cherry picked in to master branch, á la https://github.com/votinginfoproject/vip-specification/pull/372

**Comment by jdmgoogle** [comment]
Yes, please merge. Thank you.

**Pull request merged**